packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -127,3 +148,51 @@ class _ProxyOpacityProvider<T extends Object> implements OpacityProvider {
   @override
   set opacity(double value) => target.setOpacity(value, paintId: paintId);
 }
+
+class _MultiPaintOpacityProvider<T extends Object> implements OpacityProvider {
+  _MultiPaintOpacityProvider(this.paintIds, this.target) {
+    final maxOpacity = opacity;
+
+    _opacityRatios = (paintIds != null)
+        ? List<double>.generate(
+            paintIds!.length,
+            (index) =>
+                target.getOpacity(paintId: paintIds!.elementAt(index)) /
+                maxOpacity,
+          )
+        : null;
+  }
+
+  final List<T>? paintIds;
```

Comment:
```suggestion
  final List<T?>? paintIds;
```
since `null` is also a valid paint id